### DISCUSSION TITLE
Use Prettier for formatting.

### DISCUSSION THREAD

**Discussion by harisvsulaiman** [agenda/agenda#1209] [Ideas]
It looks like agenda is using xo as a linter. and doesn’t have a official stand on formatting. 
I suggest we use prettier to format the code, since it has
 
1. Reasonable defaults.
2. Editor Integration.
3. XO integration.

cc @simison @simllll @OmgImAlexis #1211

**Comment by simison** [comment]
cc @koresar and @simllll suggested the same just recently on Slack.

No objections from me! Sounds like it would make things easier and simpler to maintain for current maintainers, and fairly small task to change.

I do personally really enjoy the editor-integrations Prettier has.

@harisvsulaiman I edited your comment little bit and changed the word of choise to "reasonable". Hope you don't mind!

**Reply by harisvsulaiman** [reply]
Yeah , reasonable is the better word.🤗

**Reply by koresar** [reply]
My main problem is accessibility. I can't distinguish between ' and `

**Reply by OmgImAlexis** [reply]
> My main problem is accessibility. I can't distinguish between ' and `

What?

**Reply by koresar** [reply]
Bad eye sight

**Reply by OmgImAlexis** [reply]
I don’t understand your comment though?

**Reply by harisvsulaiman** [reply]
I guess what @koresar is saying is that double-quotes are preferable than single quotes for accessibility.

**Reply by koresar** [reply]
Yes. Thanks.

**Reply by leonardlin** [reply]
hmm, I thought prettier is something on top of xo with 'unicorn-defaults'.

I'm not the biggest fan on 'unicorn' formatting myself but I like that we have a fixed style and can stay consistent.
I'm fine with other standards, but I don't think we should focus too much on style-variations.

Like stick to one style and live with it for consistency.
Otherwise every 10th PR is just style updates.

So in essence:
If someone feels really strongly that the current 'unicorn' style is impeding his/her work, we can change.\
If it's just one or two defaults that need to change, let's just config xo accordingly.
If it's just on a 'nice to have', I would prefer to not to change the style.

@koresar can perhaps your editor highlight ' vs ` differently? VSC I think does.

**Reply by harisvsulaiman** [reply]
@leonardlin I guess thats why I proposed prettier. if accesbility is an issue we can tweak the default prettier config with “”.
As far as I’m concerned, prettier formatting defaults works fine.

**Reply by koresar** [reply]
@leonardlin yes, my IDE can highlight ' and ` differently. Now, can we teach GitHub to do that too? I review a lot of code here.

I ask for empathy.

PS
I'd like to repeat. IDE integrations with prettier are much better than xo. It's like a different level of coding.

**Comment by leonardlin** [comment]
I have nothing against that either. I have never used it but I think it makes sense.

**Comment by simison** [comment]
This is now implemented in https://github.com/agenda/agenda/pull/1211

Might be good idea to do the same in Angedash.

**Reply by koresar** [reply]
Doing it now: https://github.com/agenda/agendash/pull/165

**Comment by koresar** [comment]
Prettier is now used in both Agenda and Agendash.
Also, Travis is gone too. Both projects moved to GitHub Actions.
